sycl/source/detail/queue_impl.cpp

```diff
@@ -353,7 +338,10 @@ event queue_impl::submitMemOpHelper(const std::shared_ptr<queue_impl> &Self,
     const std::vector<event> &ExpandedDepEvents =
         getExtendDependencyList(DepEvents, MutableDepEvents, Lock);
 
-    if (areEventsSafeForSchedulerBypass(ExpandedDepEvents, MContext)) {
+    // If we have a command graph set we need to capture the op through the
+    // handler rather than by-passing the scheduler.
+    if (areEventsSafeForSchedulerBypass(ExpandedDepEvents, MContext) &&
+        !MGraph.lock()) {
```

Comment:
Nit but I would probably put the graph check first so it makes a bit more sense for short circuiting, since there's no point in calling `areEventsSafeForSchedulerBypass` if we have a graph.

Reply:
The code has been updated this way.